modules/nf-core/toulligqc/main.nf

```diff
@@ -28,18 +29,20 @@ process TOULLIGQC {
     def seq_summary_input = seq_summary ? "--sequencing-summary-source ${seq_summary}" : ""
     def fastq_input = fastq ? "--fastq ${fastq}" : ""
     def bam_input = bam ? "--bam ${bam}" : ""
+    def barcode_input = barcodes ? "--barcoding --barcodes ${barcodes}" : ""
```

Comment:
As this is jsut a string can this not just come via `$args`? It's not file based input and presumable(?) it's not required for the module to run?

Reply:
I wasn't sure about the best way to implement the barcoding option. I asked on the modules slack channel but the answer was not really clear. 

should I just create a nextfow.config file with the following?
process {

    withName: TOULLIGQC {
        ext.args = '--barcoding --barcodes barcode01,barcode02'
    }

}

Reply:
But yes, you are right it is not file based input and it is not required to run!

Reply:
Your example is exactly what you would do.

The critical thing about non-file inptu channels is they should only be an input channel is if they are _required_ for the module to run, which I guess based on the original PR, it is not?

Reply:
Okay, that makes sense! That was my initial idea but then I got confused.

Now, in terms of testing my module. How do I add the barcoding option to only one test?
for example I want to test the barcodes with only the sequencing summary file and not fastq/bam

Reply:
To one test is a good question... I guess you would want to import the module into the `test` script with teh nf-test 'alias' definition.

Then in the `nextflow.config` alongside the test file you would specify with the `alias` name of the module